app/pages/confirm.jsx

```diff
@@ -17,15 +18,59 @@ class Confirm extends React.Component {
     constructor() {
         super();
 
+        this.state = {
+            notifications: OrderedSet()
+        };
+
         this.onSubmit = this.onSubmit.bind(this);
     }
 
+    addNotification(message, key, action) {
+        return this.setState({
+            notifications: this.state.notifications.add({
+                message,
+                key,
+                action: action || 'Dismiss',
+                onClick: (notification, deactivate) => {
+                    deactivate();
+                    this.removeNotification(key);
+                },
+                dismissAfter: 5000
+            })
+        });
+    }
+
+    removeNotification(key) {
+        this.setState({
+            notifications: this.state.notifications.filter(n => n.key !== key)
+        });
+    }
+
     componentDidMount() {
         this.props.dispatch(ConfirmationThunks.loadConfirmation());
+        this.props.dispatch(ConfirmationThunks.loadForm());
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (
+            nextProps.userState.data.form &&
+            nextProps.userState.data.confirmation
+        ) {
+            for (var i in nextProps.userState.data.confirmation) {
+                if (i in nextProps.userState.data.form) {
+                    nextProps.userState.data.form[i].default =
+                        nextProps.userState.data.confirmation[i];
+                }
+            }
+        }
+
+        this.setState(nextProps);
```

Comment:
Maybe I'm missing something, but this is setting the state of this react component to the modified props and is never used, not modifying the props. Maybe it is modifying the props b/c passed by reference, but the setState call is going to set the component state to have keys for everything in nextProps.